### PR TITLE
Add component mandate for OAS resources

### PR DESCRIPTION
**Pull request by rambleraptor**
This changes our OAS guidance to ensure that the GET method returns only a schema that is annoted with `x-aep-resource`. It also changes our guidance on resource types to show that there's an expectation that all resources have a corresponding OAS component with `x-aep-resource`

## 🍱 Types of changes

What types of changes does your code introduce to AEP? _Put an `x` in the boxes
that apply_

- [x] Enhancement

## 📋 Your checklist for this pull request

Please review the [AEP Style and Guidance](https://aep.dev/style-guide) for
contributing to this repository.

### General

- [x] Basic [Guidance](https://aep.dev/style-guide#guidance) is met.
- [x] Ensure that your PR
      [references AEPs](https://aep.dev/style-guide#referencing-aeps)
      correctly.
- [x] [My code has been formatted](https://aep.dev/contributing#formatting)
      (usually `prettier -w .`)

### Additional checklist for a new AEP

- [x] A new AEP **should** be no more than two pages if printed out.
- [x] Ensure that the PR is editable by maintainers.
- [x] Ensure that [File structure](https://aep.dev/style-guide#file-structure)
      guidelines are met.
- [x] Ensure that
      [Document structure](https://aep.dev/style-guide#document-structure)
      guidelines are met.

💝 Thank you!

### CODE REVIEW

File: aep/general/0131/aep.md.j2 [hunk 1]
```diff
@@ -117,7 +117,7 @@ metadata:
 - The `operationId` **must** begin with the word `get`. The remainder of the
   `operationId` **should** be the singular form of the resource type's name.
 - The response content **must** be the resource itself. For example:
-  `#/components/schemas/Book`
+  `#/components/schemas/Book`. The response **must** reference a schema with a `x-aep-resource` extension.
```
[the remark this answers is not in the document]
Comment (rambleraptor) [reply aep/general/0131/aep.md.j2:120]: Yep! I'll do that in a follow up. Just want to make sure that the basic idea has no objections first.

File: aep/general/0004/aep.md.j2 [hunk 2]
```diff
@@ -64,19 +64,25 @@ message Topic {
 
 {% tab oas %}
 
-For OpenAPI 3.0, use the `x-aep-resource` extension:
+For OpenAPI 3.0, Resources must be defined in `#components/schemas` and use the `x-aep-resource` extension:
```
Comment (toumorokoshi) [inline aep/general/0004/aep.md.j2:67]: Some people may be hard-coding the schema into their path objects - is increasing friction for those customers a concern ?

OTOH I could imagine it's not too hard to actually transform the schema reference.

File: aep/general/0004/aep.md.j2 [hunk 3]
```diff
@@ -64,19 +64,25 @@ message Topic {
...
-  "type": "object",
-  "x-aep-resource": {
-    "singular": "user-event",
-    "plural": "user-events",
-    "patterns": [
-      "projects/{project}/user-events/{user-event}",
-      "folder/{folder}/user-events/{user-event}",
-      "users/{user}/events/{user-event}"
-    ]
+  "components": {
```
Comment (toumorokoshi) [inline aep/general/0004/aep.md.j2:71]: WDYT about updating the example to use yaml? the other ones do already and it's a bit easier to read.

File: aep/general/0004/aep.md.j2 (hunk 2 above)
Comment (rambleraptor) [reply aep/general/0004/aep.md.j2:67]: Especially if the OAS is being hand-written, I think hard-coding the full schema is an anti pattern. The multiple references will eventually get out of date. 

Using a schema is the only way I can think of to ensure users have updated all of the affected APIs.

**Pull request merged**